app/assets/src/components/views/PipelineViz.jsx

```diff
@@ -0,0 +1,366 @@
+import React from "react";
+import { DataSet, Network } from "vis";
+import { forEach } from "lodash";
+import PropTypes from "prop-types";
+import ReactPanZoom from "@ajainarayanan/react-pan-zoom";
+
+import cs from "./pipeline_viz.scss";
+
+const START_NODE_ID = -1;
+const END_NODE_ID = -2;
+const STAGE_BG_COLOR = "#f8f8f8";
+const NODE_COLOR = "#eaeaea";
+const EDGE_COLOR = "#999999";
+
+class PipelineViz extends React.Component {
+  constructor(props) {
+    super(props);
+    this.stageResults = this.props.stageResults.stages;
+    this.pipelineVersion = this.props.stageResults.pipeline_version;
+    this.stageGraphsInfo = [];
```

Comment:
since you actually push to stageGraphsInfo later, you might want to store it in the state

Reply:
Yep, changes to the graph doesn't trigger re-rendering.